sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -465,6 +520,19 @@ private WriteUnshardedBundlesToTempFiles(
               // shard numbers are assigned together to both the spilled and non-spilled files in
               // finalize.
               .apply("GroupUnwritten", GroupByKey.create())
+              .apply(
```

Comment:
Yes we rely on adding sharded keys and GBK to expand the parallelism otherwise the spilling factor will have no effect.

Reply:
Note that this is the fixed-sharding case.

Reply:
Still confused :).

Could you clarify why this was needed (since this is not in the auto-sharding path) ? Is this a bug fix for existing code ?

Reply:
This was made mainly because I changed the signature of `WriteShardsIntoTempFilesFn` to accept ((key, shard), V) to (shard, V). I reverted it in the latest commit to make the changes more clear.